concordium-consensus/src/Concordium/KonsensusV1/TreeState/Implementation.hs

```diff
@@ -774,16 +785,20 @@ finalizeTransactions = mapM_ removeTrans
     removeTrans WithMetadata{wmdData = NormalTransaction tr, ..} = do
         let nonce = transactionNonce tr
             sender = accountAddressEmbed (transactionSender tr)
-        anft <- use (transactionTable . TT.ttNonFinalizedTransactions . at' sender . non TT.emptyANFT)
-        unless (anft ^. TT.anftNextNonce == nonce) $
+        senderMinNonce <- do
+            macct <- flip getAccount (aaeAddress sender) =<< use (lastFinalized . to bpState)
+            fromMaybe minNonce <$> mapM (getAccountNonce . snd) macct
+        nextNonce <- use (transactionTable . TT.ttNonFinalizedTransactions . at' sender . non (TT.emptyANFTWithNonce senderMinNonce) . TT.anftNextNonce)
+        unless (nextNonce == nonce) $
             throwM . TreeStateInvariantViolation $
                 "The recorded next nonce for the account "
                     ++ show sender
                     ++ " ("
-                    ++ show (anft ^. TT.anftNextNonce)
+                    ++ show nextNonce
                     ++ ") doesn't match the one that is going to be finalized ("
                     ++ show nonce
                     ++ ")"
+        anft <- use (transactionTable . TT.ttNonFinalizedTransactions . at' sender . non TT.emptyANFT)
```

Comment:
This might be OK but it is a bit wonky.  In principle this anft is wrong. The next nonce is not the right one.

But as I wrote above, should this ever really be empty? Since we are finalizing the transactions in this function, they should be in the ttNonFianzliedTransactions, should they not?